lib/improver/spotdata/extract_data.py

```diff
@@ -214,10 +214,10 @@ def make_cube(self, cube, data, sites):
         """
         time_coord = cube.coord('time')
         time_coord.convert_units('hours since 1970-01-01 00:00:00')
-        latitudes = [site['latitude'] for site in sites.itervalues()]
-        longitudes = [site['longitude'] for site in sites.itervalues()]
-        utc_offsets = [site['utc_offset'] for site in sites.itervalues()]
-        site_ids = sites.keys()
+        latitudes = [float(site['latitude']) for site in sites.itervalues()]
+        longitudes = [float(site['longitude']) for site in sites.itervalues()]
+        utc_offsets = [int(site['utc_offset']) for site in sites.itervalues()]
```

Comment:
I know, but I was using a cunning bit of indexing to sort out local time zones, which won't work with half hours. As such I was basically forcing the half hour time zones to use the nearest hour. If this is entirely inappropriate I can expand out the local time zones to include half hours and fix the issue that way (but there are even 45 minute differences in places).